File: svm.scala
// import neccessary libraries
import org.apache.spark.mllib.*
import org.apache.SparkContext

//loading training data(assume it to be pre processed)
val data = MLUtils.loadLibSVMFile(sc,"/path/to/file.dat")

val altData = sc.textFile("/path/to/data.dat")

//data splitting
val splits = data.randomsplit(Array(0.6,0.4),seed=1234L)
val training = splits(0)//.cache()
val test = splits(1)

//training
val epoch = 100
val model = SVMWithSGD.tarin(training,epoch)

//computing rwa scores on test set for evaluation
val scoresAndLabels = test.map {point =>
	val score = model.predict(point.features)
	(score.point.label)
}

//get evaluation metric
val metric = new BinaryClassificationMetrics(scoresAndLabels)
val auROC = metric.areaunderROC()

println("area under ROC is :: "+ auROC)

File: svm.py
from pyspark.mllib.classification import SVMWithSGD, SVMModel
from pyspark.mllib.regression import LabeledPoint
from pyspark import SparkContext
from pyspark.mllib.util import MLUtils

# Load and parse the data
def parsePoint(line):
    values = [float(x) for x in line.split(',')]
    return LabeledPoint(values[13], values[0:4])

sc = SparkContext("local", "SVM")

#data = sc.textFile("svm_data.txt")
#parsedData = data.map(parsePoint)
trainData=MLUtils.loadLibSVMFile(sc, 'data/stripped.data')
testData=MLUtils.loadLibSVMFile(sc, 'data/stripped_test.data')

#	
#data = MLUtils.loadLibSVMFile(sc, "data/select.data", multiclass=False, numFeatures=-1, minPartitions=None)

#(trainData, testData) = data.randomSplit([0.7, 0.3])

# Build the model
model = SVMWithSGD.train(trainData, iterations=100, step=1.0, regParam=0.01, miniBatchFraction=1.0, initialWeights=None, regType='l1', intercept=False, validateData=True, convergenceTol=0.00001)
model.setThreshold(0.5)


#model=NaiveBayes.train(trainData, 1.0)

# Evaluating the model on training data
labelsAndPreds = trainData.map(lambda p: (p.label, model.predict(p.features)))
trainErr = labelsAndPreds.filter(lambda (v, p): v != p).count() / float(trainData.count())
print("Training Error = " + str(trainErr))

# Evaluating the model on test data
labelsAndPreds = testData.map(lambda p: (p.label, model.predict(p.features)))
testErr = labelsAndPreds.filter(lambda (v, p): v != p).count() / float(testData.count())
print("Test Error = " + str(testErr))



def toCSVLine(data):
  return ','.join(str(d) for d in data)

lines = labelsAndPreds.map(toCSVLine)
lines.saveAsTextFile('labels-and-predictions2.csv')

# Save and load model
model.save(sc, "myModelPath")
sameModel = SVMModel.load(sc, "myModelPath")

